### PR TITLE
add overview click-to-exit to frequent apps view

### PR DESCRIPTION
**Pull request by l3nn4rt**

### CODE REVIEW

**Comment by jderose9** [comment]
Thanks so much @l3nn4rt, missed that. Looks like this fixes 3.36 but this is still broken in 3.34 `Main.overview.viewSelector.appDisplay.views[0].view` is not an actor there. Maybe we should check `(Main.overview.viewSelector.appDisplay._views[0].view.actor || Main.overview.viewSelector.appDisplay._views[0].view)` ?

File: overview.js [hunk 1]
```diff
@@ -439,6 +439,8 @@ var dtpOverview = Utils.defineClass({
             if (activePage == ViewSelector.ViewPage.APPS) {
 
                 if(pickedActor != Main.overview._overview 
+                    && pickedActor != Main.overview.viewSelector.appDisplay._controls.get_parent()
+                    && pickedActor != Main.overview.viewSelector.appDisplay._views[0].view
```
Comment (l3nn4rt) [inline overview.js:443]:
```js
&& pickedActor != (Main.overview.viewSelector.appDisplay._views[0].view.actor || Main.overview.viewSelector.appDisplay._views[0].view)
```

File: overview.js [hunk 2]
```diff
@@ -477,32 +479,33 @@ var dtpOverview = Utils.defineClass({
...
-            ],[
-                Main.overview.viewSelector.appDisplay._views[1].view._panAction,
-                'gesture-end', 
-                Lang.bind(this, this._onAppPageSwipeEnd)
-            ]);
-        }
+        Main.overview.viewSelector.appDisplay._views.forEach((v) => {
+            if(v.view._swipeTracker) {
+                this._signalsHandler.addWithLabel('clickToExit', [
+                    v.view._swipeTracker,
```
Comment (l3nn4rt) [inline overview.js:485]:
```js
(v.view || v.view.actor)._swipeTracker,
```

File: overview.js [hunk 3]
```diff
@@ -477,32 +479,33 @@ var dtpOverview = Utils.defineClass({
...
+                    'begin',
+                    Lang.bind(this, this._onAppPageSwipeBegin)
+                ],[
+                    v.view._swipeTracker,
```
Comment (l3nn4rt) [inline overview.js:489]:
```js
(v.view || v.view.actor)._swipeTracker,
```

File: overview.js [hunk 4]
```diff
@@ -477,32 +479,33 @@ var dtpOverview = Utils.defineClass({
...
+                    'end',
+                    Lang.bind(this, this._onAppPageSwipeEnd)
+                ]);
+            } else if(v.view._panAction) {
+                this._signalsHandler.addWithLabel('clickToExit', [
+                    v.view._panAction,
```
Comment (l3nn4rt) [inline overview.js:495]:
```js
(v.view || v.view.actor)._panAction,
```

File: overview.js [hunk 5]
```diff
@@ -477,32 +479,33 @@ var dtpOverview = Utils.defineClass({
...
+                    'gesture-begin',
+                    Lang.bind(this, this._onAppPageSwipeBegin)
+                ],[
+                    v.view._panAction,
```
Comment (l3nn4rt) [inline overview.js:499]:
```js
(v.view || v.view.actor)._panAction,
```

File: overview.js [hunk 6]
```diff
@@ -477,32 +479,33 @@ var dtpOverview = Utils.defineClass({
...
+                    'gesture-cancel',
+                    Lang.bind(this, this._onAppPageSwipeEnd)
+                ],[
+                    v.view._panAction,
```
Comment (l3nn4rt) [inline overview.js:503]:
```js
(v.view || v.view.actor)._panAction,
```

**Comment by jderose9** [comment]
I made another commit to master to hopefully clean up 3.34. Thanks again!

**Pull request merged**